src/views/TimezoneModal.vue

```diff
@@ -109,53 +123,71 @@ export default defineComponent({
             text: this.$t("Confirm"),
             handler: () => {
               this.setUserTimeZone();
-            }
-          }
+            },
+          },
         ],
       });
       return alert.present();
     },
     preventSpecialCharacters($event: any) {
       // Searching special characters fails the API, hence, they must be omitted
-      if(/[`!@#$%^&*()_+\-=\\|,.<>?~]/.test($event.key)) $event.preventDefault();
+      if (/[`!@#$%^&*()_+\-=\\|,.<>?~]/.test($event.key))
+        $event.preventDefault();
     },
-    findTimeZone() { 
+    findTimeZone() {
       const queryString = this.queryString.toLowerCase();
       this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
-        return timeZone.id.toLowerCase().match(queryString) || timeZone.label.toLowerCase().match(queryString);
+        return (
+          timeZone.id.toLowerCase().match(queryString) ||
+          timeZone.label.toLowerCase().match(queryString)
+        );
       });
     },
     async getAvailableTimeZones() {
-      const resp = await UserService.getAvailableTimeZones()
-      if(resp.status === 200 && !hasError(resp)) {
+      this.showTimezoneLoading(); //function to display timezone loading modal @ashutosh7i
+      const resp = await UserService.getAvailableTimeZones();
+      if (resp.status === 200 && !hasError(resp)) {
         // We are filtering valid the timeZones coming with response here
         this.timeZones = resp.data.filter((timeZone: any) => {
           return DateTime.local().setZone(timeZone.id).isValid;
         });
         this.findTimeZone();
+        this.hideTimezoneLoading(); ////function to display timezone loading modal @ashutosh7i
       }
     },
+    // Functions to show Timezone loading modal @ashutosh7i
+    async showTimezoneLoading() {
+      const loading = await loadingController.create({
+        message: "Fetching time zones",
+      });
+      return loading.present();
+    },
+    // Functions to hide Timezone loading modal @ashutosh7i
+    async hideTimezoneLoading() {
+      await loadingController.dismiss();
+    },
+    //
     async selectSearchBarText(event: any) {
-      const element = await event.target.getInputElement()
+      const element = await event.target.getInputElement();
       element.select();
     },
     async setUserTimeZone() {
       await this.store.dispatch("user/setUserTimeZone", {
-        "timeZoneId": this.timeZoneId
-      })
-      this.closeModal()
-    }
+        timeZoneId: this.timeZoneId,
+      });
+      this.closeModal();
+    },
   },
-  beforeMount () {
+  beforeMount() {
     this.getAvailableTimeZones();
   },
   setup() {
     const store = useStore();
     return {
       close,
       save,
-      store
+      store,
     };
-  }
+  },
```

Comment:
Please revert all this unwanted changes.

https://github.com/hotwax/inventory-count/pull/215/files#diff-96ed027d1d217fd061337d2beda6aba342cdede967f533620c84c17db5e177e0R189

https://github.com/hotwax/inventory-count/pull/215/files#diff-96ed027d1d217fd061337d2beda6aba342cdede967f533620c84c17db5e177e0R181